### ISSUE TITLE
For extending the type alias created with `TypeAlias` or an assignment statement, mypy should also give error for consistency

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20520]
Memo:
- mypy test.py
- mypy 1.19.1
- Python 3.14.0

Extending the type alias created with a [type statement](https://docs.python.org/3/reference/simple_stmts.html#type) or [TypeAliasType](https://docs.python.org/3/library/typing.html#typing.TypeAliasType), mypy gives the errors as shown below:

### Type statement:

```python
type TA = int

class Cls(TA): pass # Error
```

> error: Type alias defined using "type" statement not valid as base class

### TypeAliasType:

```python
from typing import TypeAliasType

TA = TypeAliasType('TA', value=int)

class Cls(TA): pass # Error
```

> error: Variable "test.TA" is not valid as a type

But extending the type alias created with [TypeAlias](https://docs.python.org/3/library/typing.html#typing.TypeAlias) or an [assignment statement](https://docs.python.org/3/reference/simple_stmts.html#assignment-statements), mypy doesn't give error as shown below so error should also occur for consistency:

### TypeAlias:

```python
from typing import TypeAlias

TA: TypeAlias = int

class Cls(TA): pass # No error
```

### Assignment statement:

```python
TA = int

class Cls(TA): pass # No error
```

**Comment by A5rocks** [comment]
I mean, mypy is correct here.

```
>>> type TA = int
...
... class Cls(TA): pass # Error
...
Traceback (most recent call last):
  File "<python-input-0>", line 3, in <module>
    class Cls(TA): pass # Error
    ^^^^^^^^^^^^^^^^^^^
TypeError: typealias() takes exactly 2 positional arguments (3 given)
```

```
>>> from typing import TypeAliasType
...
... TA = TypeAliasType('TA', value=int)
...
... class Cls(TA): pass # Error
...
Traceback (most recent call last):
  File "<python-input-1>", line 5, in <module>
    class Cls(TA): pass # Error
    ^^^^^^^^^^^^^^^^^^^
TypeError: typealias() takes exactly 2 positional arguments (3 given)
```

The two that mypy doesn't flag don't error. I'm not sure the exact rules here and they may be wrong in some cases (which would be a bug report), but in this case mypy is matching runtime semantics and IMO that's fine.
